P1/Project_Code/frontend/src/components/Layout/Layout.jsx

```diff
@@ -1,10 +1,28 @@
-import React from "react";
-import { Outlet } from "react-router-dom";
+import { useContext, useEffect }from "react";
+import { Outlet, useNavigate } from "react-router-dom";
 import { NavigationBar } from "../Navigation/NavigationBar.jsx";
 import { SearchBar } from "../Search/SearchBar.jsx";
 import { Grid } from "@mui/material";
 
+import { UserContext } from "../../context/User.context.jsx";
+
 export const Layout = () => {
+
+  // Redirection Hook
+  const navigate = useNavigate();
+
+  // Grab Global User Data
+  const { user } = useContext(UserContext);
+
+  // Redirect to Login Route if no user
+  useEffect(() => {
+    if (!user)
+    {
+      navigate("/login");
```

Comment:
recommend moving this check logged in to own class and place in router logic
https://blog.logrocket.com/authentication-react-router-v6/
https://www.robinwieruch.de/react-router-authentication/